UAlbertaBot/UAlbertaBot_Config.txt
{
    "Bot Info" :
    {
        "BotName"                   : "UAlbertaBot",
        "Authors"                   : "David Churchill",
        "PrintInfoOnStart"          : false
    },
        
    "BWAPI" : 
    {
        "SetLocalSpeed"             : 5,
        "SetFrameSkip"              : 0,
        "UserInput"                 : true,
        "CompleteMapInformation"    : false
    },
    
    "Micro" :
    {
        "UseSparcraftSimulation"    : true,
        "KiteWithRangedUnits"       : true,
        "KiteLongerRangedUnits"     : ["Mutalisk", "Vulture"],
        "WorkersDefendRush"         : true,
        "RetreatMeleeUnitShields"   : 0,
        "RetreatMeleeUnitHP"        : 0,
        "InCombatRadius"            : 1000,
        "RegroupRadius"             : 300,
        "UnitNearEnemyRadius"       : 600
    },
    
    "Macro" :
    {
        "BOSSFrameLimit"            : 160,
        "WorkersPerRefinery"        : 3,
        "BuildingSpacing"           : 1,
        "PylonSpacing"              : 3
    },

    "Debug" :
    {
        "ErrorLogFilename"          : "bwapi-data/AI/UAlbertaBot_ErrorLog.txt",
        "LogAssertToErrorFile"      : false,
        
        "DrawGameInfo"              : false,   
        "DrawUnitHealthBars"        : true,
        "DrawProductionInfo"        : true, 
        "DrawBuildOrderSearchInfo"  : false,
        "DrawScoutInfo"             : false,
        "DrawEnemyUnitInfo"         : false,
        "DrawModuleTimers"          : false,
        "DrawResourceInfo"          : false,
        "DrawCombatSimInfo"         : false,
        "DrawUnitTargetInfo"        : false,
        "DrawBWTAInfo"              : false,
        "DrawMapGrid"               : false,
        "DrawSquadInfo"             : false,
        "DrawWorkerInfo"            : false,
        "DrawMouseCursorInfo"       : false,
        "DrawBuildingInfo"          : false,
        "DrawReservedBuildingTiles" : false,
        "DrawBOSSStateInfo"         : false,
        "PrintModuleTimeout"        : false
    },
    
    "Modules" :
    {
        "UseGameCommander"          : true,
        "UseScoutManager"           : true,
        "UseCombatCommander"        : true,
        "UseBuildOrderSearch"       : true,
        "UseStrategyIO"             : true,
        "UseAutoObserver"           : false
    },
    
    "Tools" :
    {
        "MapGridSize"               : 320
    },
    
    "Strategy" :
    {
        "Protoss"                   : "Protoss_ZealotRush",
        "Terran"                    : "Terran_MarineRush",
        "Zerg"                      : "Zerg_9D",
        
        "ScoutGasSteal"             : false,
        "ScoutHarassEnemy"          : true,
        
        "ReadDirectory"             : "bwapi-data/read/",
        "WriteDirectory"            : "bwapi-data/write/",
                
        "UseEnemySpecificStrategy"  : true,
        "EnemySpecificStrategy"     :
        {
            "BotName1"              : { "Protoss" : "Protoss_ZealotRush", "Terran" : "Terran_VultureRush", "Zerg" : "Zerg_9D" },
            "BotName2"              : { "Protoss" : "Protoss_DragoonRush", "Terran" : "Terran_MarineRush", "Zerg" : "Zerg_9D" },
            "LetaBot"               : { "Protoss" : "Protoss_ZealotRush", "Terran" : "Terran_4RaxMarines", "Zerg" : "Zerg_9D" }
        },
        
        "Strategies" :
        {
			"Zerg_Lurker_Test": { "Race" : "Zerg",    "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Extractor", "Spawning Pool", "Drone", "Drone", "Lair", "Hydralisk Den", "Lurker_Aspect", "Hydralisk", "Hydralisk", "Lurker", "Lurker"]},
            "Zerg_9D_Lurker"     : { "Race" : "Zerg",    "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Extractor", "Spawning Pool", "Drone", "Drone", "Zergling", "Zergling", "Zergling", "Lair", "Hydralisk Den"] },
            "Zerg_9D"     : { "Race" : "Zerg",    "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Spawning Pool", "Zergling", "Zergling", "Zergling", "Zergling", "Zergling", "Zergling"] }
        }
    }
}

CasiaBot/CasiaBot/CasiaBot_Config.txt
{
    "Bot Info" :
    {
        "BotName"                   : "CasiaBot",
        "Authors"                   : "David Churchill",
        "PrintInfoOnStart"          : false
    },
        
    "BWAPI" : 
    {
        "SetLocalSpeed"             : 5,
        "SetFrameSkip"              : 0,
        "UserInput"                 : true,
        "CompleteMapInformation"    : false
    },
    
    "Micro" :
    {
        "UseSparcraftSimulation"    : true,
        "KiteWithRangedUnits"       : true,
        "KiteLongerRangedUnits"     : ["Mutalisk", "Vulture"],
        "WorkersDefendRush"         : true,
        "RetreatMeleeUnitShields"   : 0,
        "RetreatMeleeUnitHP"        : 0,
        "InCombatRadius"            : 1000,
        "RegroupRadius"             : 300,
        "UnitNearEnemyRadius"       : 600
    },
    
    "Macro" :
    {
        "BOSSFrameLimit"            : 160,
        "WorkersPerRefinery"        : 3,
        "BuildingSpacing"           : 1,
        "PylonSpacing"              : 3
    },

    "Debug" :
    {
        "ErrorLogFilename"          : "bwapi-data/AI/CasiaBot_ErrorLog.txt",
        "LogAssertToErrorFile"      : false,
        
        "DrawGameInfo"              : false,   
        "DrawUnitHealthBars"        : true,
        "DrawProductionInfo"        : true, 
        "DrawBuildOrderSearchInfo"  : false,
        "DrawScoutInfo"             : false,
        "DrawEnemyUnitInfo"         : false,
        "DrawModuleTimers"          : false,
        "DrawResourceInfo"          : false,
        "DrawCombatSimInfo"         : false,
        "DrawUnitTargetInfo"        : false,
        "DrawBWTAInfo"              : false,
        "DrawMapGrid"               : false,
        "DrawSquadInfo"             : false,
        "DrawWorkerInfo"            : false,
        "DrawMouseCursorInfo"       : false,
        "DrawBuildingInfo"          : false,
        "DrawReservedBuildingTiles" : false,
        "DrawBOSSStateInfo"         : false,
        "PrintModuleTimeout"        : false
    },
    
    "Modules" :
    {
        "UseGameCommander"          : true,
        "UseScoutManager"           : true,
        "UseCombatCommander"        : true,
        "UseBuildOrderSearch"       : true,
        "UseStrategyIO"             : true,
        "UseAutoObserver"           : false
    },
    
    "Tools" :
    {
        "MapGridSize"               : 320
    },
    
    "Strategy" :
    {
        "Protoss"                   : "Protoss_ZealotRush",
        "Terran"                    : "Terran_MarineRush",
        "Zerg"                      : "Zerg_9D",
        
        "ScoutGasSteal"             : false,
        "ScoutHarassEnemy"          : true,
        
        "ReadDirectory"             : "bwapi-data/read/",
        "WriteDirectory"            : "bwapi-data/write/",
                
        "UseEnemySpecificStrategy"  : true,
        "EnemySpecificStrategy"     :
        {
            "BotName1"              : { "Protoss" : "Protoss_ZealotRush", "Terran" : "Terran_VultureRush", "Zerg" : "Zerg_9D" },
            "BotName2"              : { "Protoss" : "Protoss_DragoonRush", "Terran" : "Terran_MarineRush", "Zerg" : "Zerg_9D" },
            "LetaBot"               : { "Protoss" : "Protoss_ZealotRush", "Terran" : "Terran_4RaxMarines", "Zerg" : "Zerg_9D" }
        },
        
        "Strategies" :
        {
			"Zerg_Lurker_Test": { "Race" : "Zerg",    "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Extractor", "Spawning Pool", "Drone", "Drone", "Lair", "Hydralisk Den", "Lurker_Aspect", "Hydralisk", "Hydralisk", "Lurker", "Lurker"]},
            "Zerg_9D_Lurker"     : { "Race" : "Zerg",    "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone", "Extractor", "Spawning Pool", "Drone", "Drone", "Zergling", "Zergling", "Zergling", "Lair", "Hydralisk Den"] },
            "Zerg_9D"     : { "Race" : "Zerg",    "OpeningBuildOrder" : ["Drone", "Drone", "Drone", "Drone", "Drone"] }
        }
    }
}

CasiaBot/CasiaBot/Source/Config.cpp
#include "Config.h"
#include "CABAssert.h"

namespace Config
{
    namespace ConfigFile
    {
        bool ConfigFileFound                = false;
        bool ConfigFileParsed               = false;
		std::string ConfigFileLocation = "bwapi-data/AI/CasiaBot_Config.txt";
    }

    namespace Strategy
    {
        std::string ProtossStrategyName     = "Protoss_ZealotRush";
        std::string TerranStrategyName      = "Terran_MarineRush";
        std::string ZergStrategyName        = "Zerg_3HatchMuta";
        std::string StrategyName            = "Protoss_ZealotRush";
        std::string ReadDir                 = "bwapi-data/read/";
        std::string WriteDir                = "bwapi-data/write/";
        bool GasStealWithScout              = false;
        bool ScoutHarassEnemy               = true;
        bool UseEnemySpecificStrategy       = false;
        bool FoundEnemySpecificStrategy     = false;
    }

    namespace Modules							    
    {
        // the default tournament bot modules
        bool UsingGameCommander             = true;     // toggle GameCommander, effectively CasiaBot
        bool UsingScoutManager              = true;
        bool UsingCombatCommander           = true;
        bool UsingBuildOrderSearch          = true;     // toggle use of Build Order Search, currently no backup
        bool UsingAutoObserver              = false;
        bool UsingStrategyIO                = false;    // toggle the use of file io for strategy
        bool UsingUnitCommandManager        = false;    // handles all unit commands
		
        // extra things, don't enable unless you know what they are
        bool UsingBuildOrderDemo            = false;
    }

    namespace BotInfo
    {
        std::string BotName                 = "CasiaBot";
        std::string Authors                 = "Dave Churchill";
        bool PrintInfoOnStart               = false;
    }

    namespace BWAPIOptions
    {
        int SetLocalSpeed                   = 42;
        int SetFrameSkip                    = 0;
        bool EnableUserInput                = true;
        bool EnableCompleteMapInformation   = false;
    }
    
    namespace Tournament						
    {
        int GameEndFrame                    = 86400;	
    }
    
    namespace Debug								
    {
        bool DrawGameInfo                   = true;
        bool DrawUnitHealthBars             = true;
        bool DrawProductionInfo             = true;
        bool DrawBuildOrderSearchInfo       = false;
        bool DrawScoutInfo                  = false;
        bool DrawResourceInfo               = false;
        bool DrawWorkerInfo                 = false;
        bool DrawModuleTimers               = false;
        bool DrawReservedBuildingTiles      = false;
        bool DrawCombatSimulationInfo       = false;
        bool DrawBuildingInfo               = false;
        bool DrawMouseCursorInfo            = false;
        bool DrawEnemyUnitInfo              = false;
        bool DrawBWTAInfo                   = false;
        bool DrawMapGrid                    = false;
        bool DrawUnitTargetInfo             = false;
        bool DrawSquadInfo                  = false;
        bool DrawBOSSStateInfo              = false;
        bool PrintModuleTimeout             = false;	

        std::string ErrorLogFilename        = "CAB_ErrorLog.txt";
        bool LogAssertToErrorFile           = false;

        BWAPI::Color ColorLineTarget        = BWAPI::Colors::White;
        BWAPI::Color ColorLineMineral       = BWAPI::Colors::Cyan;
        BWAPI::Color ColorUnitNearEnemy     = BWAPI::Colors::Red;
        BWAPI::Color ColorUnitNotNearEnemy  = BWAPI::Colors::Green;
    }

    namespace Micro								
    {
        bool UseSparcraftSimulation         = true;
        bool KiteWithRangedUnits            = true;
        std::set<BWAPI::UnitType> KiteLongerRangedUnits;
        bool WorkersDefendRush              = false; 
		int RetreatMeleeUnitShields         = 0;
        int RetreatMeleeUnitHP              = 0;
        int CombatRadius                    = 1000;     // radius of combat to consider units for Micro Search
        int CombatRegroupRadius             = 300;      // radius of units around frontmost unit we consider in regroup calculation
        int UnitNearEnemyRadius             = 600;      // radius to consider a unit 'near' to an enemy unit
    }

    namespace Macro
    {
        int BOSSFrameLimit                  = 160;
        int WorkersPerRefinery              = 3;
        int BuildingSpacing                 = 1;
        int PylonSpacing                    = 3;
    }

    namespace Tools								
    {
        extern int MAP_GRID_SIZE            = 320;      // size of grid spacing in MapGrid
    }
}